src/tdsops.f90

```diff
@@ -0,0 +1,682 @@
+module m_tdsops
+   use m_common, only: dp, pi
+
+   implicit none
+
+   type :: tdsops_t
+      !! Tridiagonal Solver Operators class.
+      !! Operator arrays are preprocessed in this class based on the arguments
+      !! provided. dist_fw and dist_bw are used in the first phase of the
+      !! distributed tridiagonal solver algorithm. dist_sa and dist_sc are used
+      !! in the final substutition phase. See the kernels_dist.f90 files in the
+      !! relevant backend folders.
+      !! coeff arrays define the specific rules of building the RHS
+      !! corresponding to the tridiagonal system to be solved, and used only in
+      !! the first phase of the distributed algorithm when building the RHS.
+      !! If a boundary condition is defined then coeffs_t and coeffs_b differ
+      !! from coeffs array and define the RHS rule for the first and last 4
+      !! entries in the tridiagonal system (n_halo = 4).
```

Comment:
Could be relevant to include situations with different top / bottom level of halos. Also relevant could be situations with only one cell in a given direction. Current version of incompact3d had to deal with the case nz=1 to run 2D simulations.

Reply:
It would be good to be able to run the code with nz=1. It is a nice feature to have. We do not need to deal with it now but we would want the framework to be compatible with this feature, if possible.

Reply:
We can maybe assign the z directional operator function pointers to a null function, then we won't need to bother with n=1 and compatibility of this. If we skip the instantiation of z directional operators and never use them it shouldn't be a problem.
But in general distributed algorithm requires at least 32 points so I wasn't paying attention to what happens when n is small